editor.planx.uk/src/routes/flow.tsx

```diff
@@ -86,58 +96,69 @@ const newNode = route(async (req) => {
   };
 });
 
-const editNode = route(async (req) => {
-  const { id, before = undefined, parent = undefined } = req.params;
-
-  const node = useStore.getState().getNode(id) as {
-    type: TYPES;
-    [key: string]: any;
-  };
-
-  const extraProps = {} as any;
-
-  if (node.type === TYPES.ExternalPortal)
-    extraProps.flows = await getExternalPortals();
+// If nodeId is invalid (e.g. a deleted node), fall back to flow
+const validateEdit = (route: Matcher<object, object>) =>
+  map((req) => {
+    const { team, flow, id } = req.params;
+    const node = useStore.getState().getNode(id);
+    if (!node) return redirect(`/${team}/${flow}`);
+    return route;
+  });
```

Comment:
nit: is `validateNodeRoute` slightly more descriptive here? the original name `editNode` feels easy to confuse with graph/index methods to me, and I think this original function is really describing the action `openNode` (eg `getState()`) not the actual editing or mutating (eg click "Update <node>", call `setState()`) right?

Reply:
Very helpful comment thanks - will make the change before I merge.